specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/keys.json

```diff
@@ -0,0 +1,588 @@
+          "enum": [
+            "Purgeable",
+            "Recoverable+Purgeable",
+            "Recoverable",
+            "Recoverable+ProtectedSubscription"
+          ],
+          "x-ms-enum": {
+            "name": "DeletionRecoveryLevel",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "x-nullable": false
+        }
+      },
+      "description": "The object attributes managed by the Azure Key Vault service."
+    },
+    "KeyProperties": {
+      "properties": {
+        "attributes": {
+          "$ref": "#/definitions/KeyAttributes",
+          "description": "The attributes of the key."
+        },
+        "kty": {
+          "type": "string",
+          "minLength": 1,
+          "description": "The type of the key. For valid values, see JsonWebKeyType.",
+          "enum": [
+            "EC",
+            "EC-HSM",
+            "RSA",
+            "RSA-HSM"
+          ],
+          "x-ms-enum": {
+            "name": "JsonWebKeyType",
+            "modelAsString": true
+          }
+        },
+        "keyOps": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "The permitted JSON web key operations of the key. For more information, see JsonWebKeyOperation.",
+            "enum": [
+              "encrypt",
+              "decrypt",
+              "sign",
+              "verify",
+              "wrapKey",
+              "unwrapKey",
+              "import"
+            ],
+            "x-ms-enum": {
+              "name": "JsonWebKeyOperation",
+              "modelAsString": true
+            }
+          }
+        },
+        "keySize": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The key size in bits. For example: 2048, 3072, or 4096 for RSA."
+        },
+        "curveName": {
+          "type": "string",
+          "description": "The elliptic curve name. For valid values, see JsonWebKeyCurveName.",
+          "enum": [
+            "P-256",
+            "P-384",
+            "P-521",
+            "P-256K"
+          ],
+          "x-ms-enum": {
+            "name": "JsonWebKeyCurveName",
+            "modelAsString": true
+          }
+        },
+        "keyUri": {
+          "type": "string",
+          "description": "The URI to retrieve the current version of the key.",
+          "readOnly": true
+        },
+        "keyUriWithVersion": {
+          "type": "string",
+          "description": "The URI to retrieve the specific version of the key.",
+          "readOnly": true
+        }
+      },
+      "description": "The properties of the key."
+    },
+    "KeyAttributes": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/Attributes"
+        }
+      ],
+      "description": "The attributes of the key."
+    },
+    "KeyCreateOrUpdateParameters": {
+      "properties": {
+        "tags": {
```

Comment:
>tags [](start = 9, length = 4)

Are keys going to be tracked or proxy resources in ARM?

Reply:
The reason I ask is because tags shouldn't really be used on proxy resources.

---
In reply to: [483360952](https://github.com/Azure/azure-rest-api-specs/pull/10643#discussion_r483360952) [](ancestors = 483360952)

Reply:
Yes, keys are going to be proxy resources in ARM, in the same way that secrets are child proxy resources under vaults in ARM. See here for comparison: https://github.com/Azure/azure-rest-api-specs/blob/9603e8c3886893f114df4a2dc844d175c3f2b1c6/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/secrets.json#L361-L369


Note that Azure Key Vault's data plane supports its own tags for all objects that can be stored in a vault (keys, secrets, certificates, and managed storage account keys).

Reply:
I see.